# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/btop/hotfix-statvfs64.patch.musl
# Copyright (C) 2023 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

src/linux/btop_collect.cpp: In function 'Mem::mem_info& Mem::collect(bool)':
src/linux/btop_collect.cpp:1082:58: error: aggregate 'Mem::collect(bool)::statvfs64 vfs' has incomplete type and cannot
 be defined
 1082 |                                         struct statvfs64 vfs;
      |                                                          ^~~
src/linux/btop_collect.cpp:1083:79: error: invalid use of incomplete type 'struct Mem::collect(bool)::statvfs64'
 1083 |                                         if (statvfs64(mountpoint.c_str(), &vfs) < 0) {
      |                                                                               ^
src/linux/btop_collect.cpp:1082:48: note: forward declaration of 'struct Mem::collect(bool)::statvfs64'
 1082 |                                         struct statvfs64 vfs;
      |                                                ^~~~~~~~~

--- btop-1.2.13/src/linux/btop_collect.cpp.vanilla	2023-12-15 20:13:56.510000000 +0100
+++ btop-1.2.13/src/linux/btop_collect.cpp	2023-12-15 20:14:27.754000000 +0100
@@ -1079,8 +1079,8 @@
 				bool new_ignored = false;
 				for (auto& [mountpoint, disk] : disks) {
 					if (std::error_code ec; not fs::exists(mountpoint, ec) or v_contains(ignore_list, mountpoint)) continue;
-					struct statvfs64 vfs;
-					if (statvfs64(mountpoint.c_str(), &vfs) < 0) {
+					struct statvfs vfs;
+					if (statvfs(mountpoint.c_str(), &vfs) < 0) {
 						Logger::warning("Failed to get disk/partition stats for mount \""+ mountpoint + "\" with statvfs64 error code: " + to_string(errno) + ". Ignoring...");
 						ignore_list.push_back(mountpoint);
 						new_ignored = true;
